### PR TITLE
Display supplementary advice text on selecting end date for repeating disruptions

### PR DESCRIPTION
**Pull request by alvak1**
Navigate to "/create-disruption" page and add ending on date in disruption repeats section

### CODE REVIEW

File: site/pages/create-disruption/__snapshots__/create-disruption.test.tsx.snap [hunk 1]
```diff
@@ -1017,6 +1017,7 @@ exports[`pages > CreateDisruption > should render correctly with inputs 1`] = `
               </div>
             </fieldset>
           </div>
+          <legend />
```
Comment (RobCattonKPMG) [inline site/pages/create-disruption/__snapshots__/create-disruption.test.tsx.snap:1020]: These snapshots all seem to have empty legends - is that expected? Should we be including the new text in the snapshots?

File: site/utils/formUtils.ts [hunk 2]
```diff
@@ -66,3 +68,39 @@ export const sortStops = (stops: Stop[]) => {
...
+    disruptionStartDate?: string,
+    disruptionEndDate?: string,
+) => {
+    if (disruptionRepeats === "weekly" && disruptionStartDate && disruptionRepeatsEndDate && disruptionEndDate) {
+        const startDateDay = getFormattedDate(disruptionStartDate).day();
+        const endDateDay = getFormattedDate(disruptionEndDate).day();
+        const endingOnDateDay = getFormattedDate(disruptionRepeatsEndDate).day();
+
+        const diffInDaysWithEndingOn =
+            startDateDay < endingOnDateDay ? endingOnDateDay - startDateDay : endingOnDateDay - startDateDay + 7;
```
Comment (RobCattonKPMG) [inline site/utils/formUtils.ts:84]: As this is dealing with dates, would it be better suited to be in the dates file?

File: site/utils/formUtils.test.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { describe, it, expect } from "vitest";
+import { getEndingOnDateText } from "./formUtils";
+
+describe("formUtils tests", () => {
+    it.each([
+        ["weekly", "26/05/2023", "04/05/2023", "05/05/2023", "26/05/2023"],
+        ["weekly", "30/05/2023", "03/05/2023", "07/05/2023", "28/05/2023"],
+        ["weekly", "27/05/2023", "03/05/2023", "07/05/2023", "27/05/2023"],
+        ["weekly", "28/05/2023", "03/05/2023", "04/05/2023", "25/05/2023"],
```
Comment (RobCattonKPMG) [inline site/utils/formUtils.test.ts:9]: Should we have some scenarios where it doesnt repeat weekly?

File: site/pages/create-disruption/__snapshots__/create-disruption.test.tsx.snap (hunk 1 above)
Comment (alvak1) [reply site/pages/create-disruption/__snapshots__/create-disruption.test.tsx.snap]: done

File: site/utils/formUtils.test.ts (hunk 3 above)
Comment (alvak1) [reply site/utils/formUtils.test.ts]: done

File: site/utils/siri.ts [hunk 4]
```diff
@@ -9,14 +9,31 @@ import {
...
+import { getEndingOnDateText } from "./formUtils";
 import { Validity } from "../schemas/create-disruption.schema";
 import { Disruption } from "../schemas/disruption.schema";
 
 const getValidityPeriod = (period: Validity): Period => ({
     StartTime: getDatetimeFromDateAndTime(period.disruptionStartDate, period.disruptionStartTime).toISOString(),
     ...(period.disruptionEndDate && period.disruptionEndTime
         ? {
-              EndTime: getDatetimeFromDateAndTime(period.disruptionEndDate, period.disruptionEndTime).toISOString(),
+              EndTime:
```
[the remark this answers is not in the document]
Comment (alvak1) [reply site/utils/siri.ts]: the existing code did not have the logic to add repeating validities. This has now been added.

File: site/utils/siri.ts [hunk 5]
```diff
@@ -8,11 +8,56 @@ import {
     Period,
 } from "@create-disruptions-data/shared-ts/siriTypes";
 import dayjs from "dayjs";
-import { getDatetimeFromDateAndTime } from "./dates";
+import { getDatetimeFromDateAndTime, getFormattedDate } from "./dates";
 import { Validity } from "../schemas/create-disruption.schema";
 import { Disruption } from "../schemas/disruption.schema";
 
-const getValidityPeriod = (period: Validity): Period => ({
+const getValidityPeriod = (period: Validity): Period[] => {
```
Comment (RobCattonKPMG) [inline site/utils/siri.ts:15]: Do we have any unit tests which cover this functionality? There is quite a lot of business logic in here, slightly concerned we might not have tests for it.

Comment (alvak1) [reply site/utils/siri.ts]: done

File: site/utils/siri.test.ts [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+};
+
+const nonRepeatingPeriod: Period[] = [
+    {
+        StartTime: "2023-05-15T09:00:00.000Z",
+    },
+];
+
+describe("siri tests", () => {
+    it.each([
```
[the remark this answers is not in the document]
Comment (alvak1) [reply site/utils/siri.test.ts]: done

**Pull request merged**